src/txdb.cpp

```diff
@@ -207,6 +210,26 @@ uint256 CCoinsViewDB::GetHistoryRoot(uint32_t epochId) const {
     return root;
 }
 
+
+std::optional<libzcash::LatestSubtree> CCoinsViewDB::GetLatestSubtree(ShieldedType type) const {
```

Comment:
There is duplication of data, but I think it's worth it because it'll likely be a hot path in the consensus code, specifically for finding the block height for the latest complete subtree. We won't want to be consulting the database repeatedly for that value.